### DISCUSSION TITLE
Windows printer sample extension in store commerce app

### DISCUSSION THREAD

**Discussion by madevamgowda** [microsoft/Dynamics365Commerce.Solutions#36] [Q&A] [answered]
We had an issue in receipt alignment when we configure the printer with windows driver. we were using opos driver until now, recently we came across a requirement to print arabic text on the receipts. For some reason, arabic text not printing on the receipt when we use opos driver and started showing arabic text when we switch to windows driver. however, we come across a new issue in windows driver where part of the receipt moving towards right edge of the paper and we could see only partial receipt data. 

I understand that we may need to adjust the layout of the receipt before it reaches printer. I was referring a hardware station sample WindowsPrinterSample.cs in legacy retail sdk and I am trying to integrate the same in commerce sdk. however, some nuget packages and assemblies are not available in commerce sdk. could someone guide me if I am on right track and using the WindowsPrinterSample.cs sample fixes my issue and if so, how do I integrate the sample to commerce sdk local hardware station(using store commerce installer.

**Comment by madyke** [comment, marked as answer]
Hello @madevamgowda - there are changes in how Hardware Station extensions work in Commerce SDK, where we will provide more specific extension points rather than broad NuGet packages for customers to consume. There are a set of new samples that will show these changes that should be published soon.

You can find more details about the changes and the upcoming samples here:
[#31 - Windows printing alignment issue](https://github.com/microsoft/Dynamics365Commerce.Solutions/discussions/31)
